### ISSUE TITLE
RandomNet and degree

### ISSUE THREAD

**Issue by daniel-klein** [starsimhub/starsim#548]
RandomNet halves the desired degree for each now, and casts to integer. But this is challenging if the desired degree is 1, which will get changed to a zero, meaning no chance of infection even though the desired degree was 1.

Recommend:
```python
number_of_contacts = sc.randround(number_of_contacts / 2).astype(ss_int_)  # One-way contacts
```

This way, agents with a desired degree of 1 will get 0 or 2, but that's still better than always 0.

**Comment by cliffckerr** [comment]
I ran into this issue as well. I do wonder if we should rethink the definition of number of contacts, because it is confusing that odd numbers are basically not supported.

**Comment by daniel-klein** [comment]
I was using a Poisson distribution with a low mean, so most values were zero or one --- and was getting basically ZERO transmission!

**Comment by cliffckerr** [comment]
Fixed by #550
